WeakAuras/Transmission.lua

```diff
@@ -1659,29 +1662,68 @@ function WeakAuras.ShowDisplayTooltip(data, children, icon, icons, import, compr
   end
 end
 
-function WeakAuras.ImportString(str)
-  local received = StringToTable(str, true);
-  if(received and type(received) == "table" and received.m) then
-    if(received.m == "d") then
-      -- prepare data to be shown as a tooltip and slotted into pendingData
-      tooltipLoading = nil;
-      local data, children, icon, icons = received.d, received.c, received.i, received.a
-      WeakAuras.PreAdd(data)
-      if children then
-        for _, child in ipairs(children) do
-          WeakAuras.PreAdd(child)
+function WeakAuras.Import(inData, target)
+  local data, children, icon, icons
+  if type(inData) == 'string' then
+    -- encoded data
+    local received = StringToTable(inData, true)
+    if type(received) == 'string' then
+      -- this is probably an error message from LibDeflate. Display it.
+      ShowTooltip{
+        {1, "WeakAuras", 0.5333, 0, 1},
+        {1, received, 1, 0, 0, 1}
+      }
+      return nil, received
+    elseif received.m == "d" then
+      data = received.d
+      children = received.c
+      icon = received.i
+      icons = received.a
+    end
+  elseif type(inData.d) == 'table' then
+    data = inData.d
+    children = inData.c
+    icon = inData.i
+    icons = inData.a
+  end
+  if type(data) ~= "table" then
+    return nil, "Invalid import data."
+  end
+  local status, msg = true, ""
+  if type(target) ~= nil then
+    local targetData
+    local uid = type(target) == 'table' and target.uid or target
```

Comment:
This will always set uid to a string, so you are always iterating displays, even if passed in a target.